### PR TITLE
Rumble stats

### PR DESCRIPTION
**Pull request by Longi94**
Add rumble statistics

- Event proto for rumble items (get time, use time, player id)
- Per player and per team item stats (used, unused, average hold time)
- Added 'scored before rumble items' to Goal proto
- Added 'used item' to Goal proto (has to be active in a 3 sec time frame before the goal

TODO test cases (manually evaluated replays)
Possible edge cases to test:
- Spike broken by a freeze, the freeze has no active state (I have a replay for this)
- ???

### CODE REVIEW

File: api/metadata/game_metadata.proto [hunk 1]
```diff
@@ -37,6 +37,8 @@ message GameScore {
 message Goal {
     optional int32 frame_number = 1;
     optional api.PlayerId player_id = 2;
+    optional bool pre_rumble_items = 3;
```
Comment (dtracers) [inline api/metadata/game_metadata.proto]: We might want to wrap this in a rumble message.
That way it doesn't it take space for non rumble games.

File: api/stats/events.proto [hunk 2]
```diff
@@ -79,3 +79,11 @@ message Kickoff {
     optional int32 start_frame_number = 1;
     optional int32 end_frame_number = 2;
 }
+
+message RumbleItemEvent {
+    optional int32 frame_number_get = 1;
+    optional int32 frame_number_use = 2 [default = -1];
+    optional string item = 3;
```
Comment (dtracers) [inline api/stats/events.proto]: Can we make this an enum?

File: api/stats/player_stats.proto [hunk 3]
```diff
@@ -63,6 +64,7 @@ message PlayerStats {
     optional api.stats.Speed speed = 9;
     optional RelativePositioning relative_positioning = 10;
     optional api.stats.PerPossessionStats per_possession_stats = 11;
+    optional api.stats.RumbleItems rumble_items = 12;
```
Comment (dtracers) [inline api/stats/player_stats.proto]: Should this be rumble stats?

File: carball/analysis/stats/rumble/rumble.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+from carball.generated.api.stats.player_stats_pb2 import PlayerStats
+from carball.generated.api.stats.team_stats_pb2 import TeamStats
+from carball.generated.api.stats.events_pb2 import RumbleItemEvent
+from carball.generated.api.stats.rumble_pb2 import RumbleItems
+from carball.json_parser.game import Game
+from carball.analysis.stats.stats import BaseStat
+from carball.generated.api.metadata.game_metadata_pb2 import RANKED_RUMBLE, UNRANKED_RUMBLE
+
+POWER_UPS = [
+    'ball_freeze',
```
Comment (dtracers) [inline carball/analysis/stats/rumble/rumble.py]: If you create an enum in proto you can avoid this list and use that instead.

Comment (Longi94) [reply carball/analysis/stats/rumble/rumble.py]: make sense, thanks!

File: api/metadata/game_metadata.proto (hunk 1 above)
Comment (Longi94) [reply api/metadata/game_metadata.proto]: how did you imagine this? Maybe have something like list of pre rumble goals inside GameMetadata?

Comment (dtracers) [reply api/metadata/game_metadata.proto]: I was more imaginaging 
```
message Goal {
...
optional RumbleData rumble_data = 3;
}
```

(ofc using better naming than that)

**Comment by Longi94** [comment]
renamed few things to 'ExtraMode' like the goal info so they can include info from other extra modes too

**Comment by dtracers** [comment]
Looks like there is an issue with the proto compilation not sure if you renamed something
